spharpy/classes/ambisonics.py

```diff
@@ -0,0 +1,220 @@
+from pyfar import Signal
+from spharpy.spherical import n3d_to_maxn, n3d_to_sn3d_norm
+from spharpy.spherical import fuma_to_nm, nm_to_fuma, acn_to_nm, nm_to_acn
+import numpy as np
+
+
+class SphericalHarmonicSignal(Signal):
+    """Create audio object with spherical harmonics coefficients in time or
+    frequency domain.
+
+    Objects of this class contain spherical harmonics coefficients which are
+    directly convertible between time and frequency domain (equally spaced
+    samples and frequency bins), the channel conventions `acn` and `fuma`, as
+    well as the normalizations `n3d`, `sn3d`, or `maxn`. The default
+    parameters for `basis_type`, `normalization`, and `channel_convention`
+    corresponds to the AmbiX standard. The definition of the spherical
+    harmonics basis functions is based on the scipy convention which includes
+    the Condon-Shortley phase.
+
+
+        Parameters
+        ----------
+        data : ndarray, double
+            Raw data of the spherical harmonics signal in the time or
+            frequency domain. The memory layout of data is 'C'. E.g.
+            data of ``shape = (3, 4, 1024)`` has 3 channels with 4
```

Comment:
must the data have three dimensions or can it have any number of dimensions and only the last two must be `(..., (n_max + 1)**2, data_points)`? Should be clarified in the docstring.